From 8e31fd84cb1c7edcd897ddaaea407774de459b2e Mon Sep 17 00:00:00 2001
From: "Richard W.M. Jones" <rjones@redhat.com>
Date: Tue, 10 Sep 2013 17:25:30 +0100
Subject: [PATCH] ppc: iconv: Source is UTF-16LE not just UTF-16.

On big endian architectures like PowerPC, "UTF-16" means "UTF-16BE"!
---
 lib/utf16.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/utf16.c b/lib/utf16.c
index 4115d30..d0f2e45 100644
--- a/lib/utf16.c
+++ b/lib/utf16.c
@@ -32,7 +32,7 @@
 char *
 _hivex_windows_utf16_to_utf8 (/* const */ char *input, size_t len)
 {
-  iconv_t ic = iconv_open ("UTF-8", "UTF-16");
+  iconv_t ic = iconv_open ("UTF-8", "UTF-16LE");
   if (ic == (iconv_t) -1)
     return NULL;
 
-- 
1.8.3.1

